tests/test_swift.py

```diff
@@ -69,44 +69,64 @@ def setUp(self):
         self.channels = set()
         for appInfo in APP_INFOS.values():
             self.channels.update(appInfo.channel)
-        self.channels = sorted(self.channels)
         self.swift = swift.Swift(APP_INFOS)
 
     def test_init(self):
         self.assertEqual(self.swift.appInfos, APP_INFOS)
-        self.assertIsInstance(self.swift.mainWindow, QMainWindow)
-        self.assertIsInstance(self.swift.centralWidget, QLabel)
         for name in APP_INFOS:
             self.assertIn(name, self.swift._dockWidgets)
             self.assertIn(name, self.swift._apps)
         for channel in self.channels:
             self.assertIn(channel, self.swift._subscribers)
 
     def test_app_names(self):
-        appNames = self.swift.appNames()
-        for name in APP_INFOS:
-            self.assertIn(name, appNames)
+        appNamesSet = set(self.swift.appNames())
+        self.assertEqual(appNamesSet, set(APP_INFOS))
+
+    def test_create_app(self):
+        app_ = MagicMock()
+        app_.frames.return_value = (QWidget(),)
+        cls = MagicMock(return_value=app_)
+        setattr(importlib.import_module.return_value, "cls3", cls)
+        self.swift.createApp(
+            "app3",
+            swift.AppInfo(**{"module": "module3", "cls": "cls3", "channel": ["ch1"]})
+        )
+        self.assertIn("app3", self.swift._apps)
+        self.assertIn("app3", self.swift._dockWidgets)
+        self.assertIn("app3", self.swift._subscribers["ch1"])
 
     def test_destroy_app(self):
-        for name in APP_INFOS:
+        for name, info in APP_INFOS.items():
             self.swift.destroyApp(name)
             self.assertNotIn(name, self.swift._apps)
+            self.assertNotIn(name, self.swift._dockWidgets)
+            for channel in info.channel:
+                self.assertNotIn(name, self.swift._subscribers[channel])
 
     def test_update_frames(self):
-        self.swift._apps["app1"].frames.return_value = (QWidget(),)
+        orgFramesSet = {dockWidget.widget() for dockWidget in self.swift._dockWidgets["app1"]}
+        newFrames = (QWidget(),)
+        newFramesSet = set(newFrames)
+        self.swift._apps["app1"].frames.return_value = newFrames
         self.swift.updateFrames("app1")
+        curFramesSet = {dockWidget.widget() for dockWidget in self.swift._dockWidgets["app1"]}
```

Comment:
How about `finalFramesSet`? The term "original" and "current" look similar.

Reply:
Oh, it looks more obvious!